toggl/mobileapp
# Gain access to workspace transitions

## What's this?
Adds the sync transitions for detecting and persisting new workspaces.

### Relationships

Closes #3359 

## Why do we want this?
To persist the new workspaces.

## How is it done?
By connecting the states to the sync graph.

### Side effects
I had to 🔥 the `HasFinishedSyncBeforeInteractor` and move the logic into the sync state (which was easy btw) because a dependency loop. To create the SyncManager I needed the InteractorFactory, and to create this I needed the DataSource, and to create this I needed the SyncManager... and so on.

## Review hints
Test it!
1. Gain access to new workspaces
2. Pull to refresh
3. You should see new workspaces
4. Lose access to some workspace
5. Pull to refresh
6. Everything should work

## :squid: Permissions
Anyone, whenever
